src/Cryptography/Sodium/Bindings/SHA512.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE Trustworthy #-}
+
+-- |
+--
+-- Module: Cryptography.Sodium.Bindings.SHA512
+-- Description: Direct bindings to the SHA-512 hashing function implemented in Libsodium
+-- Copyright: (C) Hécate Moonlight 2022
+-- License: BSD-3-Clause
+-- Maintainer: The Haskell Cryptography Group
+-- Stability: Stable
+-- Portability: GHC only
+module Cryptography.Sodium.Bindings.SHA512
+  ( -- * Introduction
+    -- $introduction
+
+    -- * Single-part message
+    cryptoHashSHA512,
+
+    -- * Multi-part messages
+    CryptoHashSHA512State (..),
+    cryptoHashSHA512Init,
+    cryptoHashSHA512Update,
+    cryptoHashSHA512Final,
+    -- * Constants
+    cryptoHashSHA512Bytes,
+  )
+where
+
+import Foreign (Ptr, Word64, Word8)
+import Foreign.C (CInt (..), CSize (..), CUChar (..), CULLong (..))
+
+-- $introduction
+--
+-- The SHA-512 functions are provided for interoperability with other applications. If you are
+-- looking for a generic hash function and not specifically SHA-2, using
+-- 'Cryptography.Sodium.Bindings.GenericHashing' (BLAKE2b) might be a better choice.
+-- These functions are also not suitable for hashing passwords or deriving keys from passwords.
+-- Use 'Cryptography.Sodium.Bindings.PasswordHashing' instead.
+--
+-- These functions are not keyed and are thus deterministic. In addition, the untruncated versions
+-- are vulnerable to length extension attacks. A message can be hashed in a single pass, but a
+-- streaming API is also available to process a message as a sequence of multiple chunks.
+
+
+-- | Hash the content of the @in@ buffer and put the result in the @out@ buffer.
+-- The @inlen@ parameter is the size of what is in the @in@ buffer.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_hash_sha512"
+  cryptoHashSHA512 ::
+    -- | @out@ parameter. This is a pointer to the hash of your data.
```

Comment:
I don't think the C parameter names need to be in the docs. Users of the bindings want to know what the parameters are for, not what another language they're deliberately not interacting with chooses to name them.

Reply:
Ok!